plugin/advisor/catalog/catalog.go

```diff
@@ -19,6 +19,12 @@ type Database struct {
 	Collation    string
 	DbType       db.Type
 	SchemaList   []*Schema
+
+	// AllowMissing defines the policy for presence detection.
+	// In some cases we can not fetch the catalog, such as GitHub App/Actions.
+	// It's hard to distinguish these cases from exactly empty database.
+	// So we need allowMissing to distinguish them.
+	AllowMissing bool
```

Comment:
My feeling is this field is like context info thus would be more appropriate to be inside databaseState.

And I may consider reversing the logic and naming it checkCatalog.

The reason is:

1. Avoid negation
2. checkCatalog is an extra capability.
3. More nature in the code `if checkCatalog` {...}